pymeasure/instruments/keysight/keysightDSOX1102G.py

```diff
@@ -0,0 +1,559 @@
+        100, 250, 500, 1000, 2 000, 5 000, 10 000, 20 000, 50 000, 62 500.
+        
+        Note that the oscilloscope may provide less than the specified nb of points. """,
+        validator=strict_discrete_set,
+        values=[100, 250, 500, 1000, 2000, 5000, 10000, 20000, 50000, 62500]
+    )
+    waveform_source = Instrument.control(
+        ":waveform:source?", ":waveform:source %s",
+        """ A string parameter that selects the analog channel, function, or reference waveform 
+        to be used as the source for the waveform methods. Can be "channel1", "channel2", "function", 
+        "fft", "wmemory1", "wmemory2", or "ext".""",
+        validator=strict_discrete_set,
+        values={"channel1": "CHAN1", "channel2": "CHAN2", "function": "FUNC", "fft": "FFT",
+                "wmemory1": "WMEM1", "wmemory2": "WMEM2", "ext": "EXT"},
+        map_values=True
+    )
+    waveform_format = Instrument.control(
+        ":waveform:format?", ":waveform:format %s",
+        """ A string parameter that controls how the data is formatted when sent from the 
+        oscilloscope. Can be "ascii".""",
+        validator=strict_discrete_set,
+        values={"ascii": "ASC"},
+        map_values=True
+    )
+    @property
+    def waveform_preamble(self):
+        """ Get preamble information for the selected waveform source as a dict with the following keys:
+            - "format": byte, word, or ascii (str)
+            - "type": normal, peak detect, or average (str)
+            - "points": nb of data points transferred (int)
+            - "count": always 1 (int)
+            - "xincrement": time difference between data points (float)
+            - "xorigin": first data point in memory (float)
+            - "xreference": data point associated with xorigin (int)
+            - "yincrement": voltage difference between data points (float)
+            - "yorigin": voltage at center of screen (float)
+            - "yreference": data point associated with yorigin (int)"""
+        return self._waveform_preamble()
+
+    @property
+    def waveform_data(self):
+        """ Get the binary block of sampled data points transmitted using the IEEE 488.2 arbitrary
+        block data format."""
+        # Other waveform formats raise UnicodeDecodeError
+        self.waveform_format = "ascii"
+
+        data = self.values(":waveform:data?")
+        # Strip header from first data element
+        data[0] = float(data[0][10:])
+        return data
+
+    ################
+    # System Setup #
+    ################
+
+    @property
+    def system_setup(self):
+        """ A string parameter that sets up the oscilloscope. Must be in IEEE 488.2 format.
+        It is recommended to only set a string previously obtained from this command."""
+        return self.ask(":system:setup?")
+
+    @system_setup.setter
+    def system_setup(self, setup_string):
+        self.write(":system:setup " + setup_string)
+
+    def __init__(self, adapter, **kwargs):
```

Comment:
can you please move the init to the top of the class? It's maybe personal preference, but I find it a little weird to bury this in the middle somewhere. I know others in the codebase are also like this, I should formalize this one day.